addons/api/addon/models/host-catalog.js

```diff
@@ -2,9 +2,42 @@ import GeneratedHostCatalogModel from '../generated/models/host-catalog';
 
 export default class HostCatalogModel extends GeneratedHostCatalogModel {
   /**
+   * Returns if the host-catalog is static type or not.
    * @type {boolean}
    */
   get isStatic() {
     return this.type === 'static';
   }
+
+  /**
+   * Returns if the host-catalog is a plugin or not.
+   * @type {boolean}
+   */
+  get isPlugin() {
+    return this.type === 'plugin';
+  }
+
+  /**
+   * Returns if host-catalog plugin is AWS or not
+   * @type {boolean}
+   */
+  get isAWS() {
+    return this.plugin.name === 'aws';
+  }
+
+  /**
+   * Return if host-catalog plugin is Azure or not.
+   * @type {boolean}
+   */
+  get isAzure() {
+    return this.plugin.name === 'azure';
+  }
+
+  /** If host-catalog is a plugins returns its name,
+   * otherwise returns the host-catalog type
+   * @type {string}
+   */
+  get compositeType() {
+    return this.isPlugin ? this.plugin.name : this.type;
+  }
```

Comment:
Let's add a `set compositeType(type)` field that handles setting values of `static` or `*`.

- If setting this field to `static`, the underlying field `type` should be set to `static`.
- For any other value, `type` should be set to `plugin` and `plugin.name` should be set to the value.